graphql/server.go

```diff
@@ -344,8 +387,17 @@ func Handler(schema *Schema) http.Handler {
 	})
 }
 
+func (c *conn) Use(fn MiddlewareFunc) {
+	c.middlewares = append(c.middlewares, fn)
+}
+
 func ServeJSONSocket(ctx context.Context, socket JSONSocket, schema *Schema, makeCtx MakeCtxFunc, logger GraphqlLogger) {
```

Comment:
Can you clarify the second point? I find the create/use thing a little strange because we're creating/hooking up middleware once per every connection handler.

I left `ServeJSONSocket` for compatibility, but maybe it's fine to break.